src/main/java/ch/fhnw/algd1/converters/utf8/UTF8Converter.java

```diff
@@ -1,15 +1,41 @@
 package ch.fhnw.algd1.converters.utf8;
 
+import java.util.ArrayList;
+
 /*
  * Created on 05.09.2014
  */
 /**
- * @author 
+ * @author
  */
 public class UTF8Converter {
 	public static byte[] codePointToUTF(int x) {
 		byte[] b = null;
+		int firstStep = 0x80;
+		int secondStep = 2047;
+		int thirdStep = 0x10000;
+		int lastStep = 0x200000;
 		// TODO allocate b in the right size (depending on x) and fill it with the
+		if (x < firstStep) {
+			b = new byte[1];
+			b[0] = (byte) x;
+		} else if (x < secondStep) {
+			b = new byte[2];
+			b[0] = (byte) ((x >>> 6 | 0xC0));
+			b[1] = (byte) ((x & 0x3F) | 0x80);
+		} else if (x < thirdStep) {
+			b = new byte[3];
+			b[0] = (byte) (0xE0 | x >>> 12);
+			b[1] = (byte) (0x80 | x >>> 6);
```

Comment:
Hier (und beim letzten Fall an zwei solchen Orten) fehlt jetzt noch die gleiche Maske wie in der nächsten Zeile, damit die Ergebnisse korrekt sind.